examples/policy_gradient/reinforce/agent.py

```diff
@@ -2,48 +2,258 @@
+          tf.stack(values=[t.policy.mean_kl for t in self.towers]), 0)
+      self.mean_entropy = tf.reduce_mean(
+          tf.stack(values=[t.policy.mean_entropy for t in self.towers]), 0)
+
+    # References to the model weights
+    self.variables = ray.experimental.TensorFlowVariables(
+        self.common_policy.loss,
+        self.sess)
+    self.observation_filter = MeanStdFilter(preprocessor.shape, clip=None)
+    self.reward_filter = MeanStdFilter((), clip=5.0)
     self.sess.run(tf.global_variables_initializer())
 
+  def setup_common_policy(self, observations, advantages, actions, prev_log):
+    with tf.variable_scope("tower"):
+      self.common_policy = ProximalPolicyLoss(
+          self.env.observation_space, self.env.action_space,
+          observations, advantages, actions, prev_log, self.logit_dim,
+          self.kl_coeff, self.distribution_class, self.config, self.sess)
+
+  def setup_per_device_policy(
+          self, device, observations, advantages, actions, prev_log):
+    with tf.device(device):
+      with tf.variable_scope("tower", reuse=True):
+        all_obs = tf.Variable(
+            observations, trainable=False, validate_shape=False,
+            collections=[])
+        all_adv = tf.Variable(
+            advantages, trainable=False, validate_shape=False, collections=[])
+        all_acts = tf.Variable(
+            actions, trainable=False, validate_shape=False, collections=[])
+        all_plog = tf.Variable(
+            prev_log, trainable=False, validate_shape=False, collections=[])
+        obs_slice = tf.slice(
+            all_obs,
+            [self.batch_index] + [0] * len(self.preprocessor.shape),
+            [self.per_device_batch_size] + [-1] * len(self.preprocessor.shape))
+        obs_slice.set_shape(observations.shape)
+        adv_slice = tf.slice(
+            all_adv, [self.batch_index], [self.per_device_batch_size])
+        acts_slice = tf.slice(
+            all_acts,
+            [self.batch_index] + [0] * len(self.action_shape),
+            [self.per_device_batch_size] + [-1] * len(self.action_shape))
+        plog_slice = tf.slice(
+            all_plog, [self.batch_index, 0], [self.per_device_batch_size, -1])
+        policy = ProximalPolicyLoss(
+            self.env.observation_space, self.env.action_space,
+            obs_slice, adv_slice, acts_slice, plog_slice, self.logit_dim,
+            self.kl_coeff, self.distribution_class, self.config, self.sess)
+        grads = self.optimizer.compute_gradients(
+            policy.loss, colocate_gradients_with_ops=True)
+
+      return Tower(
+          tf.group(
+              *[all_obs.initializer,
+                all_adv.initializer,
+                all_acts.initializer,
+                all_plog.initializer]),
+          grads,
+          policy)
+
+  def load_data(self, trajectories, full_trace):
+    """
+    Bulk loads the specified trajectories into device memory. The data is
+    split equally across all the devices.
+
```

Comment:
There should be a short one line summary after the """

Also, the return documentation should be formatted as

```
Returns:
  The number of tuples loaded per device.
```